src/main/java/com/marek/onlinebookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -29,15 +29,15 @@ public class ShoppingCartServiceImpl implements ShoppingCartService {
     @Override
     public ShoppingCartDto findUserCart(User user) {
         return shoppingCartMapper.toDto(shoppingCartRepository
-                .findShoppingCartByUser(user));
+                .findShoppingCartByUser(user.getId()));
     }
 
     @Override
     public ShoppingCartDto addItem(User user, CartItemRequestDto cartItemRequestDto) {
         CartItem item = cartItemMapper.toModel(cartItemRequestDto);
         item.setBook(bookRepository.findById(item.getBook().getId())
                 .orElseThrow(EntityNotFoundException::new));
-        ShoppingCart cart = shoppingCartRepository.findShoppingCartByUser(user);
+        ShoppingCart cart = shoppingCartRepository.findShoppingCartByUser(user.getId());
```

Comment:
Similar to the findShoppingCartByUser method, the user should be checked if it is null before getting its id and passing it to the method. This is to avoid any NullPointerExceptions.